### ISSUE TITLE
Random lava popping sound when exiting to menu

### ISSUE THREAD

**Issue by Ved-s** [bluemagic123/tModLoader#2429]
### Version

Latest 1.4-preview

### OS

Windows

### Platform

Steam

### This bug affects

Gameplay as a Player

### Description

Just as title explains, lava popping sound may randomly start playing on exiting the worl

https://user-images.githubusercontent.com/53968411/170172583-4c5d9809-9e09-429c-91d7-16555d238a06.mp4
(took 8-10 attempts to reproduce)

### Log File

[client.log](https://github.com/tModLoader/tModLoader/files/8767646/client.log)

### Steps to reproduce

The issue appears completely random on exiting to menu

### Expected Behavior

It should play MenuClose on exit

### Actual Behavior

It randomly plays lava popping sound (I don't know its ID)

### Reproduction frequency

10-15% chance

**Comment by Ouroel** [comment]
can confirm

**Comment by dragon3025** [comment]
I'm also having this issue in v2022.05.103.9, without any mods enabled.

**Comment by Ved-s** [comment]
Also it looks like that sound and music get disabled after some time

**Comment by AxerTheAxe** [comment]
I would also like to mention that you can overlap the sound multiple times by leaving and re-joining a world and the sound that plays when you open settings from your inventory is cut off and does not play normally.

**Comment by JavidPack** [comment]
fixed a while ago

"Yeah, I found and fixed it a while ago
TL;DR someone added code to WorldGen’s save/save&exit callbacks to cut off the waterfall and lavafall sounds, but it broke when the sound API changed" -absoluteAquarian
